spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/Mqttv5PahoMessageHandler.java

```diff
@@ -73,16 +75,34 @@ public class Mqttv5PahoMessageHandler extends AbstractMqttMessageHandler
 
 	public Mqttv5PahoMessageHandler(String url, String clientId) {
 		super(url, clientId);
-		this.connectionOptions = new MqttConnectionOptions();
-		this.connectionOptions.setServerURIs(new String[]{ url });
-		this.connectionOptions.setAutomaticReconnect(true);
+		this.connectionOptions = buildDefaultConnectionOptions(url);
 	}
 
 	public Mqttv5PahoMessageHandler(MqttConnectionOptions connectionOptions, String clientId) {
 		super(obtainServerUrlFromOptions(connectionOptions), clientId);
 		this.connectionOptions = connectionOptions;
 	}
 
+	public Mqttv5PahoMessageHandler(ClientManager<IMqttAsyncClient> clientManager,
+			MqttConnectionOptions connectionOptions) {
```

Comment:
Same here: the `MqttConnectionOptions` msut be controlled by that `ClientManager`.
Probably if we would implement all of these from scratch, we would go with only `ClientManager` abstraction and no any simple `url + clientId`.

Reply:
Yeah I would also remove it, the only problem is that `connectionOptions` is used in `getConnectionInfo` methods.
1. It can be theoretically generilized by adding a new method in the `ClientManager` interface, like `getConnectionProperties` and introducing a new generics type so that now this interface will have to be `ClientManager<T, P>` where `P` is `ConnectionOptions` class which is different for v3 and v5.
2. We can simply return empty connection options instance in case `clientManager` is present.

Reply:
See my comment in other place on the matter: the `ClientManager` could just extend `MqttComponent`.